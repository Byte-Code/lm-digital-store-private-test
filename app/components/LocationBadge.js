import React, { PropTypes } from 'react';
import styled from 'styled-components';

import IconSelector from './Icons';
import DateBadge from './DateBadge';

const Wrapper = styled.div`
  display: flex;
  width: 100%;
  padding: 0 40px 70px;
  &>div {
    width: 50%;
  }
`;

const LeftColumn = styled.div`
  display: flex;
  align-items: flex-end;
`;

const InfoWrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: 48px;
  justify-content: flex-end;
`;

const Location = styled.p`
  font-size: 32px;
  font-family: LeroyMerlinSans;
  color: #fff;
`;

const Temp = styled.div`
  text-align: right;
  color: #fff;
  font-family: LeroyMerlinSans Bold;
  font-size: 160px;
  line-height: 0.94;
`;

const LocationBadge = ({ city, country, weather, temp }) => (
  <Wrapper>
    <LeftColumn>
      <IconSelector weather={weather} />
      <InfoWrapper>
        <Location>{city} ({country})</Location>
        <DateBadge />
      </InfoWrapper>
    </LeftColumn>
    <Temp>{temp} &#176;</Temp>
  </Wrapper>
);

LocationBadge.propTypes = {
  city: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired,
  weather: PropTypes.string.isRequired,
  temp: PropTypes.number.isRequired
};

export default LocationBadge;
