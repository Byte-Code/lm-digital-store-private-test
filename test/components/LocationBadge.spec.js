import React from 'react';
import { shallow } from 'enzyme';

import LocationBadge from '../../app/components/LocationBadge';

describe('LocationBadge', () => {
  it('renders properly', () => {
    const result = shallow(
      <LocationBadge
        city="milano"
        country="Italia"
        weather="sunny"
        temp={30}
      />
    );
    expect(result).toMatchSnapshot();
  });
});
