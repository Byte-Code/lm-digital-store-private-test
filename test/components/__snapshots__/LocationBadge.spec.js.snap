// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationBadge renders properly 1`] = `
<styled.div>
  <styled.div>
    <IconSelector
      weather="sunny"
    />
    <styled.div>
      <styled.p>
        milano
         (
        Italia
        )
      </styled.p>
      <DateBadge />
    </styled.div>
  </styled.div>
  <styled.div>
    30
     Â°
  </styled.div>
</styled.div>
`;
